saw-core/src/Verifier/SAW/FiniteValue.hs

```diff
@@ -66,22 +66,25 @@ data FirstOrderType
   | FOTArray FirstOrderType FirstOrderType
   | FOTTuple [FirstOrderType]
   | FOTRec (Map FieldName FirstOrderType)
-  deriving (Eq, Show, Generic)
+  deriving (Eq, Ord, Show, Generic)
 
 -- | Values inhabiting those first-order types.
 -- NB: The JSON encoding of this type, used for saw-script solver result caching,
 -- assumes constructor names and argument orders will not change (though the
 -- order and number of constructors may change) - see 'firstOrderJSONOptions'
+--
+-- The type argument of FOVArray is the key type; the value type is
+-- derivable from the default value, which is the second argument.
```

Comment:
Make sure to describe what the significance of the `Map` argument is as well (either here or in `FOVArray`-specific Haddocks).